x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalVerifier.java

```diff
@@ -23,6 +24,12 @@ public Failures verify(LogicalPlan plan) {
         Failures failures = new Failures();
         Failures dependencyFailures = new Failures();
 
+        // AwaitsFix https://github.com/elastic/elasticsearch/issues/118531
+        var enriches = plan.collectFirstChildren(Enrich.class::isInstance);
+        if (enriches.isEmpty() == false && ((Enrich) enriches.get(0)).mode() == Enrich.Mode.REMOTE) {
+            return failures;
+        }
```

Comment:
This is following the pattern for the physical verifier. It exempts queries with remote enriches from dependency checking in both the local and global logical optimizers. That's okay.

But maybe we could do a little better:
- We only need to exclude the remote enriches from the dependency checks, not the whole verification. (When https://github.com/elastic/elasticsearch/pull/118534 excluded remote enriches, the verifier used to check fewer things - it's doing more things now than just check dependencies.)
- By making another instance of the verifier specifically for the local optimizer, we could have only the local instance skip remote enriches.

If we go for these, we should align the physical verifier as they generally work the same.